server/apikeys.go

```diff
@@ -0,0 +1,138 @@
+package main
+
+import (
+	"errors"
+	"log/slog"
+	"net/http"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"gorm.io/gorm"
+)
+
+type ApiKey struct {
+	ID        uint      `gorm:"primarykey" json:"id"`
+	CreatedAt time.Time `json:"createdAt"`
+	Value     string    `gorm:"not null" json:"value"`
+	UserID    uint      `gorm:"not null" json:"-"`
+	Note      string    `gorm:"not null" json:"note"`
+}
+
+type ApiKeyRequest struct {
+	Note string `json:"note,required"`
+}
+
+func (b *BaseRouter) addAPIKeyRoutes() {
+	apiKeyGrp := b.rg.Group("/apikeys").Use(AuthRequired(nil))
```

Comment:
I think `/apikeys` here could be `/api_keys`. Just to keep it inline with the naming of other routes.